src/plugins/controllerboard2x/index.js

```diff
@@ -1,70 +1,88 @@
-      if (enable){
-        deps.globalEventLoop.emit('mcu.SendCommand', 'escp(1)');
-      } else {
-        deps.globalEventLoop.emit('mcu.SendCommand', 'escp(0)');      
-      }
-    });
+var sharedFunctions = require('./public/js/telemetryToSystemPower.js');
+var PREFERENCES_NS = 'plugins';
 
-  };
-  SystemPower.prototype.getSettingSchema = function getSettingSchema() {
-    //Technically the json-editor supports "watch" that can be used
-    //to dynamically add items from an array as an ENUM to a select
-    //in the same document.  Have an issue filed as I cannot get it
-    //to work.  WORKAROUND: Manually parse the config for the Battery
-    //types and inject them in to the schema.  TODO: signal an update
-    //to the schema that is caches in several spots when a Battery
-    //is changed.
-    BatteryOptions = [
-      'TrustFire',
-      'LiFePO4'
-    ];
-    var b = this.config.preferences.get(PREFERENCES_NS);
-    if ('batteryDefintions' in b) {
-      b.batteryDefintions.batteries.forEach(function (bat) {
-        BatteryOptions.push(bat.name);
+class SystemEnvironment
+{
+    constructor(name, deps)
+    {
+      console.log('Controllerboard2x:SystemEnvironment plugin loaded');
+    }
+}
+
+class SystemPower
+{
+    constructor(name, deps)
+    {
+      console.log('Controllerboard2x:SystemPower plugin loaded');
+
+      this.config   = deps.config;
+      this.cockpit  = deps.cockpit;
+      var self      = this;
+
+      deps.cockpit.on('plugin.systemPower.powerESCs', function (enable) 
+      {
+        if (enable)
+        {
+          deps.globalEventLoop.emit('mcu.SendCommand', 'escp(1)');
+        } 
+        else 
+        {
+          deps.globalEventLoop.emit('mcu.SendCommand', 'escp(0)');      
+        }
       });
     }
-    return [{
-        'id': 'batteryDefintions',
-        'title': 'Battery Formulas',
-        'category' : 'hardware',        
-        'type': 'object',
-        'properties': {
+
+    getSettingSchema()
+    {
+      // Technically the json-editor supports "watch" that can be used
+      // to dynamically add items from an array as an ENUM to a select
+      // in the same document.  Have an issue filed as I cannot get it
+      // to work.  
+      
+      // WORKAROUND [REMOVED, SEE NOTE]: Manually parse the config for 
+      // the Battery types and inject them in to the schema.
+      // [NOTE]: Replaced with static configuration.
+
+      // TODO: signal an update to the schema that is caches in several
+      // spots when a Battery is changed.
+
+      var BatteryOptions = [
+        'TrustFire',
+        'LiFePO4 (OpenROV White)',
+        'NMC (OpenROV Blue)'
+      ];
+
+      return [{
+        'id':       'batteryDefinitions',
+        'title':    'Battery Formulas',
+        'category': 'hardware',        
+        'type':     'object',
+
+        'properties': 
+        {
           'batteries': {
-            'id': 'batteries',
+            'options': 
```

Comment:
Added hidden option so that the battery list doesn't show up. I don't think users should be able to edit this.